assets/src/domain/eventEditor/ui/ticketAssignmentsManager/useTAMStateManager.ts

```diff
@@ -0,0 +1,70 @@
+import { useEffect } from 'react';
+
+import { EntityId } from '@appServices/apollo/types';
+import { useRelations } from '@appServices/apollo/relations';
+import useAssignmentManager from './useAssignmentManager';
+import useValidateTAMData from './useValidateTAMData';
+import { TAMStateManager, AssignmentStatus } from './types';
+
+const useTAMStateManager = (): TAMStateManager => {
+	const { initialize, ...assignmentManager } = useAssignmentManager();
+
+	// The existing relations to be used to create initial data
+	// and to calculate difference between new and old data
+	const relations = useRelations();
+	const orphanEntities = useValidateTAMData(assignmentManager);
+
+	useEffect(() => {
+		// initialize with existing data
+		initialize(relations.getData());
+	}, []);
+
+	const hasNoAssignedDates = ({ ticketId }) => orphanEntities.tickets.includes(ticketId);
+
+	const hasNoAssignedTickets = ({ datetimeId }) => orphanEntities.datetimes.includes(datetimeId);
+
+	const hasOrphanTickets = () => orphanEntities.tickets.length > 0;
+
+	const hasOrphanDates = () => orphanEntities.datetimes.length > 0;
+
+	const hasOrphanEntities = () => hasOrphanTickets() || hasOrphanDates();
+
+	const getOldRelation = ({ datetimeId }): EntityId[] => {
```

Comment:
fun times eh?

really impressed with how you have been building up logic like this from smaller pieces